plugins/modules/aci_epg_subnet.py

```diff
@@ -0,0 +1,477 @@
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+SUBNET_CONTROL_MAPPING = dict(
+    nd_ra="nd",
+    no_gw="no-default-gateway",
+    querier_ip="querier",
+    unspecified="",
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        epg=dict(type="str", aliases=["epg_name", "name"]),  # Not required for querying all objects
+        ap=dict(type="str", aliases=["app_profile", "app_profile_name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        enable_vip=dict(type="bool"),
+        gateway=dict(type="str", aliases=["gateway_ip"]),  # Not required for querying all objects
+        mask=dict(type="int", aliases=["subnet_mask"]),  # Not required for querying all objects
+        subnet_name=dict(type="str", aliases=["name"]),
+        nd_prefix_policy=dict(type="str"),
+        preferred=dict(type="bool"),
+        route_profile=dict(type="str"),
+        route_profile_l3out=dict(type="str"),
+        scope=dict(type="list", elements="str", choices=["private", "public", "shared"]),
+        subnet_control=dict(type="str", choices=["nd_ra", "no_gw", "querier_ip", "unspecified"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        ip_data_plane_learning=dict(type="str", choices=["enabled", "disabled"], aliases=["ip_dataplane_learning"]),
+        tenant=dict(type="str", aliases=["tenant_name"]),  # Not required for querying all objects
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_together=[["gateway", "mask"], ["route_profile_l3out", "route_profile"]],
+        required_if=[
+            ["state", "present", ["epg", "ap", "gateway", "mask", "tenant"]],
+            ["state", "absent", ["epg", "ap", "gateway", "mask", "tenant"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    description = module.params.get("description")
+    enable_vip = aci.boolean(module.params.get("enable_vip"))
+    epg = module.params.get("epg")
+    ap = module.params.get("ap")
+    tenant = module.params.get("tenant")
+    gateway = module.params.get("gateway")
+    mask = module.params.get("mask")
+    if mask is not None and mask not in range(0, 129):
+        # TODO: split checks between IPv4 and IPv6 Addresses
+        module.fail_json(msg="Valid Subnet Masks are 0 to 32 for IPv4 Addresses and 0 to 128 for IPv6 addresses")
+    if gateway is not None:
+        gateway = "{0}/{1}".format(gateway, str(mask))
```

Comment:
format - will take care of the type conversion.
```suggestion
        gateway = "{0}/{1}".format(gateway, mask)
```